Form/ChoosePaymentMethodType.php

```diff
@@ -92,15 +92,18 @@ protected function buildChoiceList(FormBuilderInterface $builder, array $options
             $label = 'form.label.'.$method;
 
             if (Legacy::formChoicesAsValues()) {
-                $options['choices'][$method] = $label;
-            } else {
                 $options['choices'][$label] = $method;
+                if (Legacy::needsChoicesAsValuesOption()) {
+                    $options['choices_as_values'] = true;
+                }
+            } else {
+                $options['choices'][$method] = $label;
```

Comment:
Here is the code for earlier than 2.7. Choices are not flipped. Otherwise, they are, and `choices_as_values` is set to true when available.